public/app/features/alerting/AlertRuleItem.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import Highlighter from 'react-highlight-words';
+import classNames from 'classnames/bind';
+import { togglePauseAlertRule } from './state/actions';
+import { AlertRule } from '../../types';
+
+export interface Props {
+  rule: AlertRule;
+  search: string;
+  togglePauseAlertRule: typeof togglePauseAlertRule;
+}
+
+class AlertRuleItem extends PureComponent<Props, any> {
+  togglePaused = () => {
+    const { rule } = this.props;
+
+    this.props.togglePauseAlertRule(rule.id, { paused: rule.state !== 'paused' });
```

Comment:
Shouldn't this just callback to parent component to make this a "dumb" component?